### ISSUE TITLE
origin store wrong when doing payments

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2064]
_Imported from trac ticket 2064.
Created by **sal**
Opened in trac: Jan 11, 2013, 9:23:36 AM
Last modified in trac: Jan 17, 2013, 12:47:50 PM_
***
_by sal:_
There seems to be a mismatch of meaning of origin store between retailers. I think we keep "fixing" this back and forth. Need to investigate whether there's a GERS system option we're not looking at.

-Rob
___

See below. Did something change with update?

Sent from my iPhone

Begin forwarded message:

From: "Sharon Ford" <slf@ivansmith.com>
Date: December 22, 2012 12:49:33 PM CST
To: "'Sal Genovese'" <sal@ivansmith.com>
Subject: RE: LOFT ORIGIN STORE PAYMENTS

It’s not taking payments that’s the problem – It’s using the store # that took the payment as the origin store # – I had 6 accounts to fix from 12/21/12:
 
LEE9N47572        11242A6GL52     Store 03 took the money for Store A6  sale and the origin store for the transaction had 03 which is the payment cash drawer store                                              
GREEM48420     1019201AFIV      Store 05 took the money for Store 01 sale and the origin store for the transaction had 05 which is the payment cash drawer store
67275                    1219216OB93     Store 09 took the money for Store 16 sale and the origin store for the transaction had 09 which is the payment cash drawer store
 
Sharon Ford
Accounting Supervisor
Ivan Smith Furniture
5434 Technology Drive
Shreveport, LA  71129
 
(318) 683-3908; fax:  (318) 688-0399
Email:  slf@ivansmith.com
 
From: Sal Genovese [mailto:sal@ivansmith.com] 
Sent: Saturday, December 22, 2012 12:22 PM
To: Sharon Ford
Subject: Re: LOFT ORIGIN STORE PAYMENTS
 
No that was fixed long time ago for taking payments but we did do an update Wednesday and that may have broke something. Can u send me one specific example?

Sent from my iPhone

On Dec 22, 2012, at 9:42 AM, "Sharon Ford" <slf@ivansmith.com> wrote:

Sal
 
Can’t remember, but did you say that the fix was in for payments taken at a store for another store?  Friday 12/21/12 several payments still have the wrong origin store for payments that cross locations?  Loft is still using the cash drawer as the origin store – Where are we on this?
 
Thanks 
 
Sharon Ford
Accounting Supervisor
Ivan Smith Furniture
5434 Technology Drive
Shreveport, LA  71129
 
(318) 683-3908; fax:  (318) 688-0399
Email:  slf@ivansmith.com

**Comment by kentbower** [comment]
_**rneville** trac comment on Jan 14, 2013, 4:35:14 PM:_

```
protected function artranDefaults(tran:TransferObject):void {
            tran.cashdrawer = SessionAuth.cashdrawerid;
            tran.sourcesiteid = SessionAuth.siteid;		
        }
```

in PaymentPM.as looks like the culprit here. This shouldn't be defaulted from the client because all of the code on the server side only goes through it's logic if the value is None...we need to let the server determine this and not the client.

**Comment by kentbower** [comment]
_**rneville** trac comment on Jan 17, 2013, 12:47:50 PM:_

(Changeset [[3559]](../commit/cc7ba11572cba9b7d2b732667ba85f5ba2de1063))
* fixes #2064 (origin store wrong when doing payments)
 * fixes #2040 (origin store wrong when doing a dollar transfer)
 * fixes #2010 (dollar transfer origin store wrong in certain situation and DTC/DTD pair shouldn't record a paymentsiteid)
 * DEVELOPER NOTE: No client logic should ever set sourcesiteid for an ArTran. The server will figure that part out as there's too many business rules involved to leave it at the client layer.
 * DEVELOPER NOTE: Cleaned up ArTra.transactiondefaults method to accept a kwargs {} so that calls to it are easier to understand the side-effects of how fields on the resulting ArTran will be populated.
 * remove unused TG controllers and templates

**Comment by kentbower** [comment]
_**rneville** trac status change on Jan 17, 2013, 12:47:50 PM:_ **closed**
